### PR TITLE
Book Routes

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
New routes for Api wbooks (Index and Show).
### ScreenShoots
image
### Trello
https://trello.com/c/0RfYarKw/22-crear-rutas-index-y-show-para-book
### Notes
/api/v1/books for Index (All books)
/api/v1/books/(id) for a specific book.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,4 +29,7 @@
 # Folders ignored
 test/
 
+# VSCode
```
Comment (facucompe) [inline .gitignore:32]: Add this file to `.gitignore`

File: app/assets/javascripts/books.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (facucompe) [inline app/assets/javascripts/books.coffee]: Why did you add this file

File: app/assets/stylesheets/books.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Books controller here.
```
Comment (facucompe) [inline app/assets/stylesheets/books.scss]: Same here

File: app/helpers/books_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module BooksHelper
```
Comment (facucompe) [inline app/helpers/books_helper.rb]: Remove this file

File: app/views/books/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1>Books#show</h1>
```
Comment (facucompe) [inline app/views/books/show.html.erb]: Same for this

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1 do
+     resources :books
```
Comment (facucompe) [inline config/routes.rb]: You only define an index an a show, resources defines many other routes. You can do 
`resources :books, only: [:show, :index]`

File: app/controllers/api/v1/books_controller.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
+      before_action :authenticate_user!
+      # GET /books
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: Remove comments

**Review by facucompe — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (mcolombo87) [reply .gitignore:32]: I don't get it, ".vscode" is just there.

File: app/assets/javascripts/books.coffee (hunk 2 above)
Comment (mcolombo87) [reply app/assets/javascripts/books.coffee]: It was auto generated by rails g controllers...

File: app/assets/stylesheets/books.scss (hunk 3 above)
Comment (mcolombo87) [reply app/assets/stylesheets/books.scss]: It was auto generated by rails g controllers...

File: app/views/books/show.html.erb (hunk 5 above)
Comment (mcolombo87) [reply app/views/books/show.html.erb]: Done!

File: app/helpers/books_helper.rb (hunk 4 above)
Comment (mcolombo87) [reply app/helpers/books_helper.rb]: Done!

File: config/routes.rb (hunk 6 above)
Comment (mcolombo87) [reply config/routes.rb]: Done!.

File: app/controllers/api/v1/books_controller.rb (hunk 7 above)
Comment (mcolombo87) [reply app/controllers/api/v1/books_controller.rb]: Done!

File: Gemfile [hunk 8]
```diff
@@ -41,6 +41,13 @@ gem 'devise', '~> 4.5.0'
 gem 'devise_token_auth', '~> 1.0.0'
 gem 'omniauth'
 
+# For Models to jsons
+gem 'active_model_serializers', '~> 0.10.0'
+
+# For Pagination
+gem 'wor-paginate'
```
Comment (facucompe) [inline Gemfile]: Lock version.

File: .gitignore (hunk 1 above)
Comment (facucompe) [reply .gitignore]: It should not be in the repository, it's a file for your editor. I haven't seen it until now

File: app/assets/javascripts/books.coffee (hunk 2 above)
Comment (facucompe) [reply app/assets/javascripts/books.coffee]: Ok, but we won't have views. Remove it

File: app/assets/stylesheets/books.scss (hunk 3 above)
Comment (facucompe) [reply app/assets/stylesheets/books.scss]: Same here. Remove it

File: app/controllers/api/v1/books_controller.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
+      before_action :authenticate_user!
+      def index
+        render_paginated Book
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: You install serializer gem but you are not serializing the records

**Review by facucompe — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (facucompe) [reply .gitignore:32]: Sorry, hahah, i thougth it was another file

File: app/assets/stylesheets/books.scss (hunk 3 above)
Comment (mcolombo87) [reply app/assets/stylesheets/books.scss]: Done

File: Gemfile (hunk 8 above)
Comment (mcolombo87) [reply Gemfile]: Done

File: app/controllers/api/v1/books_controller.rb (hunk 9 above)
Comment (mcolombo87) [reply app/controllers/api/v1/books_controller.rb]: Done

**Review by facucompe — APPROVED** [review]
